File: services/events-service/src/main/java/org/prism/eventsservice/model/EventRequest.java
package org.prism.eventsservice.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.Instant;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@AllArgsConstructor
public class EventRequest {
    @JsonProperty("event_key")
    private String eventKey;

    @JsonProperty("user_details")
    private UserDetails userDetails;

    @JsonProperty("experiment_details")
    private ExperimentDetails experimentDetails;

    @JsonProperty("sent_at")
    private Instant sentAt;

    private Map<String, Object> properties;
}

File: tools/experiment-simulator/internal/model/eventRequest.go
package model

import "time"

type EventRequest struct {
	EventKey          EventKey           `json:"event_key"`
	UserDetails       UserDetails        `json:"user_details"`
	ExperimentDetails ExperimentDetails  `json:"experiment_details"`
	SentAt            time.Time          `json:"sent_at"`
	Properties        map[EventField]any `json:"properties"`
}

type ExperimentDetails struct {
	ExperimentKey string `json:"experiment_key"`
	VariantKey    string `json:"variant_key"`
}

type UserDetails struct {
	Id string `json:"id"`
}
